ICLIP/iclip_demultiplex.py

```
@@ -0,0 +1,92 @@
+    res = resdk.Resolwe(args.username, args.password, args.server)
+    resdk.start_logging()
+
+    # Upload input data
+    multiplexed_reads = res.run('upload-fastq-single', input={'src': args.multiplexed_reads})
+    sample_annotation = res.run('upload-iclip-annotation', input={'src': args.sample_annotation})
+
+    # trigger demultiplexing, annotation and primary analysis
+    demultiplex = res.run(
+        'workflow-icount-demultiplex',
+        input = {
+            'reads': multiplexed_reads.id, # reference previously uploaded data by its ID
+            'icount_annotation': sample_annotation.id # reference previously uploaded data by its ID
+        }
+    )
+
+    while demultiplex.status not in ['OK', 'ER']:
+        sleep(5)
+        demultiplex.update()
+
+    if demultiplex.status == 'OK':
+
+        demux_object = res.data.filter(parents=demultiplex.id, type='data:demultiplex:icount:')[0]
+        annotation_object = res.data.filter(parents=demultiplex.id, type='data:icount:annotsample:')[0]
+
+        # wait for sample annotation job to either finish or fail
+        while annotation_object.status not in ['OK', 'ER']:
+            sleep(5)
+            annotation_object.update()
+
+        if annotation_object.status == 'OK':
+            # get a list of child objects after demultiplexing
+            demux_data = res.data.filter(parents=demux_object.id)
+
+            # Use experiment_name annotation field to assign samples to collection
+            for d_obj in demux_data:
+                if d_obj.sample:
+                    try:
+                        collection_name = d_obj.descriptor['other']['experiment_name']
+                        coll = get_or_create_collection(res, collection_name)
+
+                        if coll:
+                            coll.add_samples(d_obj.sample)
+                            print('{} added to collection {}.'.format(d_obj.sample, coll.name))
+                    except:
```

In what kind of scenario we expect this to fail? If there is a predicted error, we could use something more specific than bare except statement...